ffi/capi/src/units_converter.rs

```diff
@@ -0,0 +1,77 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use icu_experimental::units::converter::UnitsConverter;
+use icu_experimental::units::converter_factory::ConverterFactory;
+use icu_experimental::units::measureunit::MeasureUnit;
+
+#[diplomat::bridge]
+pub mod ffi {
+    use crate::errors::ffi::ICU4XError;
+    use alloc::boxed::Box;
+    use core::str;
+
+    #[diplomat::opaque]
+    /// An ICU4X Units Converter Factory object, capable of creating converters a [`ICU4XUnitsConverter`]
+    /// from [`ICU4XMeasureUnit`]s.
+    /// Also, it can parse the CLDR unit identifier (e.g. `meter-per-square-second`) and get the [`ICU4XMeasureUnit`].
+    #[diplomat::rust_link(icu::experimental::units::converter_factory::ConverterFactory, Struct)]
+    #[diplomat::rust_link(
+        icu::experimental::units::converter_factory::ConverterFactory,
+        Struct,
+        hidden
+    )]
+    pub struct ICU4XUnitsConverterFactory(pub ConverterFactory);
+
+    impl ICU4XUnitsConverterFactory {
+        /// Creates a new [`ICU4XUnitsConverterFactory`] from supplemental data
+        // (TODO: is it locale data? or something else?)
+        #[diplomat::rust_link(
+            icu::experimental::units::converter_factory::ConverterFactory::new,
+            FnInStruct
+        )]
+        pub fn create() -> Result<Box<ICU4XUnitsConverterFactory>, ICU4XError> {
+            Ok(Box::new(ICU4XUnitsConverterFactory(call_constructor!(
```

Comment:
issue: this should have a data provider argument on `create` and passed as the last argument to `call_constructor!`

I'm surprised this compiles: do ConverterFactory's data loading methods need tweaks?

Reply:
@robertbastian  and @Manishearth 

I would like to have help on this.

Now, the ConverterFactory takes a units data provider that is not depending on the locale, and I want to pass it to the constructor here.

Reply:
You need to do something like https://github.com/unicode-org/icu4x/blob/b4952328c8207dbd77793c93923b3fb71b5a983a/ffi/capi/src/casemap.rs#L59

Reply:
Thanks, done